library/threading.c

```diff
@@ -40,10 +40,11 @@ static void threading_mutex_init_pthread( mbedtls_threading_mutex_t *mutex )
 
 static void threading_mutex_free_pthread( mbedtls_threading_mutex_t *mutex )
 {
-    if( mutex == NULL )
+    if( mutex == NULL || !mutex->is_valid )
```

Comment:
I like this part of the fix, but I am wondering whether we need a similar check in `threading_mutex_init_pthread()`. The standard says:
```
Attempting to initialise an already initialised mutex results in undefined behaviour. 
```
However, I think this is basically protecting the library from itself, the best solution is to actually fix multiple initialisation/destroy calls.